File: src/main/java/io/github/mg138/tsbook/command/admin/AdminCommands.java
package io.github.mg138.tsbook.command.admin;

import io.github.mg138.tsbook.Book;

import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.jetbrains.annotations.NotNull;

public class AdminCommands implements CommandExecutor {
    protected static String ITEM;

    public static String getItem() {
        return ITEM;
    }

    public static void setItem(String s) {
        ITEM = s;
    }

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull org.bukkit.command.Command command, @NotNull String label, @NotNull String[] args) {
        ITEM = null;

        switch (args.length) {
            case 0:
                sender.sendMessage(Book.Companion.getCfg().translate.translate("messages.help"));
                return true;

            case 1:
                switch (args[0].toLowerCase()) {
                    case "reload":
                        return Reload.call(sender);
                    case "help":
                        return Help.call(sender);
                    case "get":
                        return Get.call(sender);
                    case "give":
                        return Give.call(sender);
                    case "unid":
                        return Unidentified.call(sender);
                    case "identify":
                        return Identify.call(sender);
                    case "effect":
                        return Effect.call(sender);
                }

            case 2:
                switch (args[0].toLowerCase()) {
                    case "get":
                        return Get.call(sender, args[1]);
                    case "give":
                        return Give.call(sender);
                    case "unid":
                        return Unidentified.call(sender);
                    case "effect":
                        return Effect.call(sender);
                }
            case 3:
                switch (args[0].toLowerCase()) {
                    case "give":
                        return Give.call(sender, args[1], args[2]);
                    case "unid":
                        return Unidentified.call(sender, args[1], args[2]);
                    case "effect":
                        if (args[2].equals("clear")) return Effect.clear(sender, args[1]);
                        return Effect.call(sender);
                }
            case 4:
                switch (args[0].toLowerCase()) {
                    case "effect":
                        return Effect.call(sender);
                }
            case 5:
                switch (args[0].toLowerCase()) {
                    case "effect":
                        return Effect.call(sender, args[1], args[2], args[3], args[4]);
                }
        }

        sender.sendMessage(Book.Companion.getCfg().translate.translate("errors.unknown_command"));
        return false;
    }
}

File: src/main/java/io/github/mg138/tsbook/command/admin/AdminCommands.kt
package io.github.mg138.tsbook.command.admin

import io.github.mg138.tsbook.setting.BookConfig
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

class AdminCommands : CommandExecutor {
    companion object {
        var item: String? = null
    }

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        item = null
        val category = args[0].toLowerCase()
        when (args.size) {
            0 -> {
                sender.sendMessage(BookConfig.translate.translate("messages.help"))
                return true
            }
            1 -> when (category) {
                "reload" -> return Reload.call(sender)
                "help" -> return Help.call(sender)
                "get" -> return Get.call(sender)
                "give" -> return Give.call(sender)
                "unid" -> return Unidentified.call(sender)
                "identify" -> return Identify.call(sender)
                "effect" -> return Effect.call(sender)
            }
            2 -> when (category) {
                "get" -> return Get.call(sender, args[1])
                "give" -> return Give.call(sender)
                "unid" -> return Unidentified.call(sender)
                "effect" -> return Effect.call(sender)
            }
            3 -> when (category) {
                "give" -> return Give.call(sender, args[1], args[2])
                "unid" -> return Unidentified.call(sender, args[1], args[2])
                "effect" -> return when (args[2]) {
                    "clear" -> Effect.clear(sender, args[1])
                    else -> Effect.call(sender)
                }
            }
            4 -> when (category) {
                "effect" -> return Effect.call(sender)
            }
            5 -> when (category) {
                "effect" -> return Effect.call(sender, args[1], args[2], args[3], args[4])
            }
        }
        sender.sendMessage(BookConfig.translate.translate("errors.unknown_command"))
        return false
    }
}